### PR TITLE
Allow host to provide a shader map.

### PR DESCRIPTION
**Pull request by dj2**
This CL extends the Amber interface to allow a Shader map to be
provided. This map allows the caller to pre-compile the shaders and
provide the name to data mapping. The ShaderCompiler will then pull from
the map before attempting to do a compile.

### CODE REVIEW

File: include/amber/amber.h [hunk 1]
```diff
@@ -15,12 +15,16 @@
...
 
+#include <map>
 #include <string>
+#include <vector>
 
 #include "amber/result.h"
 
 namespace amber {
 
+using ShaderMap = std::map<std::string, std::vector<uint32_t>>;
```
Comment (jaebaek) [inline include/amber/amber.h]: `ShaderMap` is not intuitive to me. Could you please add a comment to explain meaning of its key and value?

Comment (dj2) [reply include/amber/amber.h]: done

Comment (jaebaek) [reply include/amber/amber.h]: nit: represntation --> representation

File: include/amber/amber.h [hunk 2]
```diff
@@ -48,6 +54,13 @@ class Amber {
   /// Executes the given |data| script with the provided |opts|. Returns a
   /// |Result| which indicates if the execution succeded.
   amber::Result Execute(const std::string& data, const Options& opts);
+
+  /// Executes the given |data| script with the provided |opts. Will use
```
Comment (jaebaek) [inline include/amber/amber.h]: nit: |opts --> |opts|

File: include/amber/amber.h (hunk 1 above)
Comment (dj2) [reply include/amber/amber.h]: done

File: include/amber/amber.h (hunk 2 above)
Comment (dj2) [reply include/amber/amber.h]: done

**Pull request merged**